modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,464 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Encode, Decode, HasCompact};
+use frame_support::{Parameter, decl_error, decl_event, decl_module, decl_storage, ensure, traits::{Get, EnsureOrigin, FungibleAsset, MintableAsset, BurnableAsset}};
+use frame_system::{ensure_none, ensure_signed};
+use edgeware_primitives::Balance;
+use sp_core::ecdsa;
+use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
+use sp_runtime::{
+	Permill,
+	ModuleId,
+	traits::{Member, StaticLookup, AccountIdConversion},
+	transaction_validity::{
+		InvalidTransaction, TransactionPriority, TransactionSource, TransactionValidity, ValidTransaction,
+	},
+	DispatchResult,
+};
+use sp_std::vec::Vec;
+use coinaddress as btc_address;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+type EcdsaSignature = ecdsa::Signature;
+type DestAddress = Vec<u8>;
+
+
+
+type TokenIdOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::AssetId;
+type BalanceOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::Balance;
+
+const NAME_MAX_LENGTH : u8 = 32;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type RenVMTokenIdType: Member + Parameter + Default + Copy + HasCompact;
+	type RenvmBridgeUnsignedPriority: Get<TransactionPriority>;
+	type ControllerOrigin: EnsureOrigin<Self::Origin>;
+	type ModuleId: Get<ModuleId>;
+	type Assets: FungibleAsset<Self::AccountId> + MintableAsset<Self::AccountId> + BurnableAsset<Self::AccountId>;
+}
+
+/// struct				RenTokenInfo
+/// ren_token_id		What this pallet uses to identify tokens ( unique for a given token on this pallet for a given chain )
```

Comment:
Not sure I undestand what this parameter is for, just match them up with the assets ID on assets pallet.

Reply:
This ren_token_id parameter is separate from the id used by the assets pallet, so ren_token_id(s) of renToken as seen by the users is sequential. We can do without it, I'll merge ren_token_id and  ren_token_asset_id.